### PR TITLE
feat: add skip-test-tls flag to optionally skip testing for tls (#9679)

### LINKED ISSUE #9679: argocd login just hangs on 2.4.0
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Running `argocd login <server> --sso` does nothing and just hangs. The exact same command works on 2.3.

**To Reproduce**

* Download the 2.4.0 binary (https://github.com/argoproj/argo-cd/releases/tag/v2.4.0).
* Run `argocd login <server-name> --sso`.
* It doesn't seem like it's related to my specific Argo instance, because it even hangs if I run something like `argocd login random.com --sso`.
* Adding `--loglevel debug` doesn't show anything either.

**Expected behavior**

A browser tab is opened to go through the SSO login flow.

**Screenshots**

N/A

**Version**

```shell
argocd: v2.4.0+91aefab
  BuildDate: 2022-06-10T17:38:43Z
  GitCommit: 91aefabc5b213a258ddcfe04b8e69bb4a2dd2566
  GitTreeState: clean
  GoVersion: go1.18.3
  Compiler: gc
  Platform: darwin/amd64
```

**Logs**

N/A

### PR DESCRIPTION
**Pull request by msvechla**
This issue fixes #9679 by implementing a new CLI flag `--skip-test-tls`, which optionally skips checking whether the server is configured with tls.

As mentioned in the related issue, adding this flag is probably the best solution here: https://github.com/argoproj/argo-cd/issues/9679#issuecomment-1185752420

My company is blocked from updating the argocd-cli at the moment, same as many others using AWS ELB in the setup. This MR at least gives us a way forward by using the new flag.

Let me know if I need to make any further adjustments.

Thanks and best regards!

Signed-off-by: msvechla <m.svechla@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by msvechla** [comment]
I rebased again, can someone please take a look?

**Comment by msvechla** [comment]
@crenshaw-dev can you take a look maybe, as you also mentioned this as a preferred option here: https://github.com/argoproj/argo-cd/issues/9679#issuecomment-1185752420

File: cmd/argocd/commands/login.go
Comment (crenshaw-dev) [inline cmd/argocd/commands/login.go:74]: This is awesome! Two suggestions:

1. Does this need to be a global config option, or can we _just_ add it to `login`?
2. How would you feel about adding `(this can help when the command hangs for no apparent reason)` or something like that so people can discover the option without finding the related GitHub issue?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by crenshaw-dev** [comment]
Also I'd be comfortable marking this as a fix instead of a feature and backporting it to 2.4. @leoluz can you give a second opinion on that?

File: cmd/argocd/commands/login.go
Comment (msvechla) [reply cmd/argocd/commands/login.go]: Hi, you are completely right:

1. moved the new flag to the login sub-command
2. added the hint about when this flag might help

**Comment by msvechla** [comment]
@crenshaw-dev thanks for the input, I addressed your comments. Let me know if I should change the commit to `fix` instead of `feat`

**Review by crenshaw-dev — APPROVED** [review]
lgtm! I'm going to merge and backport. If another maintainer disagrees, I'll just undo the backport.  :-D

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.12.

**Pull request merged**
